src/ast.rs

```diff
@@ -127,7 +150,7 @@ pub struct BinaryExpr {
 
 impl Compile for BinaryExpr {
     fn compile(&self, t: &Target) -> String {
-        unimplemented!()
+        format!("\"%{}%\" {} \"%{}%\"", self.left.compile(t), self.op.compile(t), self.right.compile(t))
```

Comment:
You don't want the % to be here. The % would be added when a variable is compiled (hence why you're compiling both sides. Remove the quotes as well.